azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.container;
+
+import static azkaban.ServiceProvider.SERVICE_PROVIDER;
+
+import azkaban.AzkabanCommonModule;
+import azkaban.Constants;
+import azkaban.Constants.PluginManager;
+import azkaban.db.DBMetrics;
+import azkaban.db.DatabaseOperator;
+import azkaban.db.MySQLDataSource;
+import azkaban.execapp.AzkabanExecutorServer;
+import azkaban.execapp.ExecMetrics;
+import azkaban.execapp.FlowRunner;
+import azkaban.execapp.event.FlowWatcher;
+import azkaban.execapp.event.RemoteFlowWatcher;
+import azkaban.executor.ActiveExecutingFlowsDao;
+import azkaban.executor.AssignExecutorDao;
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.ExecutionFlowDao;
+import azkaban.executor.ExecutionJobDao;
+import azkaban.executor.ExecutionLogsDao;
+import azkaban.executor.ExecutionOptions;
+import azkaban.executor.ExecutionRampDao;
+import azkaban.executor.ExecutorDao;
+import azkaban.executor.ExecutorEventsDao;
+import azkaban.executor.ExecutorLoader;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.executor.FetchActiveFlowDao;
+import azkaban.executor.JdbcExecutorLoader;
+import azkaban.executor.NumExecutionsDao;
+import azkaban.executor.Status;
+import azkaban.jobtype.HadoopJobUtils;
+import azkaban.jobtype.HadoopProxy;
+import azkaban.jobtype.JobTypeManager;
+import azkaban.metrics.CommonMetrics;
+import azkaban.metrics.MetricsManager;
+import azkaban.project.JdbcProjectImpl;
+import azkaban.project.ProjectLoader;
+import azkaban.security.commons.HadoopSecurityManager;
+import azkaban.server.AzkabanServer;
+import azkaban.utils.Props;
+import azkaban.utils.StdOutErrRedirect;
+import azkaban.utils.Utils;
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.zip.ZipFile;
+import javax.sql.DataSource;
+import org.apache.commons.dbutils.QueryRunner;
+import org.apache.log4j.Logger;
+
+/**
+ *  This class is responsible for launching a flow in a container.
```

Comment:
Can you please add more detail in documentation for this class? It has main logic to start a flow container in containerized environment.